src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -267,44 +290,74 @@ def onMouseClickOnImage(self, x: float, y: float):
         indices = self._map_plot_widget.getImageIndices(x, y)
         if indices is None:
             return
         radial_background, pattern_background = self.getBackgroundCurve(row=indices.row, col=indices.col)
 
-        # Unset the background if it's the same pixel
-        if self._intensity_background_curve:
-            if (indices.row, indices.col) in self._intensity_background_curve.keys():
-                self._intensity_background_curve = {}
-                markers = [item for item in self._map_plot_widget.getItems() if isinstance(item, MarkerBase)]
-                _ = [self._map_plot_widget.removeMarker(legend=marker.getLegend()) for marker in markers if "BG" in marker.getLegend()]
-
-                for legend, intensity_curve in self._intensity_curves_cache.items():
-                    self._integrated_plot_widget.addCurve(
-                        x=radial_background,
-                        y=intensity_curve,
-                        legend=legend,
-                        selectable=False,
-                        color=self._integrated_plot_widget.getCurve(legend=legend).getColor()
-                    )
-                return
+        # Unset the background if it's the same pixel and delete markers
+        if (indices.row, indices.col) in self._intensity_background_curve.keys():
+            self._intensity_background_curve = {}
+            markers = [item for item in self._map_plot_widget.getItems() if isinstance(item, MarkerBase)]
+            _ = [self._map_plot_widget.removeMarker(legend=marker.getLegend()) for marker in markers if "BG" in marker.getLegend()]
+        else:
+            self._intensity_background_curve = {(indices.row, indices.col) : pattern_background}
 
-        if radial_background is not None and pattern_background is not None:
+        if self._intensity_background_curve:
             self._intensity_background_curve = {(indices.row, indices.col) : pattern_background}
             pixel_center_coords = self._map_plot_widget.findCenterOfNearestPixel(x, y)
             self._map_plot_widget.addMarker(
                 *pixel_center_coords, color="black", symbol="x", legend=f"BG_LOCATION"
             )
 
-        self.subtractBackground(radial_background, pattern_background)
+        # Substitute non-fixed curve
+        for indices, intensity_curve in self._intensity_curve_cache.items():
+            intensity_curve_ = copy.copy(intensity_curve)
+            if self._intensity_background_curve:
+                intensity_curve_ -= pattern_background
+
+            self._integrated_plot_widget.addCurve(
+                    x=radial_background,
+                    y=intensity_curve_,
+                    legend="INTEGRATE",
+                    selectable=False,
+                    color=self._integrated_plot_widget.getCurve(legend="INTEGRATE").getColor(),
+                    resetzoom=self._integrated_plot_widget.getGraphXLimits() == (0,100),
```

Comment:
Why is the rationale behind this `resetzoom` condition ?

Reply:
it's the original shape, so if the user changes it, the zoom will not reset (of course, there is a better way to do this)

Reply:
Doesn't seem to work on my end. I guess it is data-dependant ?

There is probably a programatic way in silx to know if the zoom was changed or not.

Reply:
Like: `len(PlotWidget.getLimitsHistory()) == 0`